lib/apiservers/engine/backends/volume.go

```diff
@@ -58,7 +118,62 @@ func (v *Volume) VolumeRm(name string) error {
 		if _, ok := err.(*storage.RemoveVolumeConflict); ok {
 			return derr.NewRequestConflictError(fmt.Errorf("Volume is in use"))
 		}
-		return derr.NewErrorWithStatusCode(fmt.Errorf("Server error form portlayer"), http.StatusInternalServerError)
+		return derr.NewErrorWithStatusCode(fmt.Errorf("Server error from portlayer: %s", err), http.StatusInternalServerError)
 	}
 	return nil
 }
+
+//Utility Functions
+
+func fillDockerVolumeModel(volume *models.VolumeResponse, labels map[string]string) *types.Volume {
+	dockerVol := types.Volume{
+		Driver:     volume.Driver,
+		Name:       volume.Name,
+		Labels:     labels,
+		Mountpoint: volume.SelfLink,
+	}
+	return &dockerVol
+}
+
+func validateDriverArgs(args map[string]string) error {
+	//volumestore name validation
+	if _, ok := args[optsVolumeStoreKey]; !ok {
+		args[optsVolumeStoreKey] = "default"
+	}
+
+	//capacity validation
+	if _, ok := args[optsCapacityKey]; !ok {
```

Comment:
why not write this as below instead of two accesses to the map?:
```
cap, ok := args[optsCapacityKey]
if !ok {
  ...
  return nil
}

if _, convErr := ...

```